pkg/actor/actor.go

```diff
@@ -50,13 +52,34 @@ type Actor interface {
 	Act(context.Context, *resource.Cluster) error
 }
 
+// NewOperatorActions creates a slice of actors that control the actions or actors for the operator.
+// The order of the slice is critical so that the actors run in order, for instance update has to
+// happen before deploy.
 func NewOperatorActions(scheme *runtime.Scheme, cl client.Client, config *rest.Config) []Actor {
-	return []Actor{
+
+	var update Actor
+
+	// entry point for new PartitionUpdate upgrades
+	// this feature is controlled by a featuregate
+	if utilfeature.DefaultMutableFeatureGate.Enabled(features.PartitionedUpdate) {
+		update = newPartitionedUpdate(scheme, cl, config)
+	} else {
+		update = newUpgrade(scheme, cl, config)
+	}
+
+	// The order of these actors MATTERS.
+	// We need to have update before deploy so that
+	// updates run before the deploy actor, or
+	// deploy will update the STS container and not
+	// deploy.
+	actor := []Actor{
```

Comment:
nit: `actors`

Reply:
actually, we did not need the variable.  I removed it.